tests/jballerina-semtype-test/src/test/java/io/ballerina/test/SemTypeTest.java

```diff
@@ -0,0 +1,130 @@
+            // This check is used to remove failing tests.
+            if (!file.getName().contains("-f.bal")) {
+                files.add(file.getAbsolutePath());
+            }
+        }
+        return files.toArray();
+    }
+
+    @Test(dataProvider = "filePathProvider")
+    public void testSubTypeRelationship(String filePath) throws IOException {
+        List<String> actualRelations = actualSubTypeRelations(filePath);
+        List<String> expectedRelations = expectedSubTypeRelations(filePath);
+        Assert.assertEquals(actualRelations.size(), expectedRelations.size());
+        for (String actualRelation : actualRelations) {
+            expectedRelations.removeIf(expectedRelation -> Objects.equals(actualRelation, expectedRelation));
+        }
+        Assert.assertEquals(expectedRelations.size(), 0);
+    }
+
+    private List<String> actualSubTypeRelations(String filePath) {
+        CompileResult compileResult = BCompileUtil.compile(filePath);
+        BLangPackage bLangPackage = (BLangPackage) compileResult.getAST();
+        List<BTypeDefinitionSymbol> bTypeDefinitionSymbols = new ArrayList<>();
+        for (Scope.ScopeEntry value : bLangPackage.symbol.scope.entries.values()) {
+            BSymbol bSymbol = value.symbol;
+            if (bSymbol.kind == SymbolKind.TYPE_DEF) {
+                bTypeDefinitionSymbols.add((BTypeDefinitionSymbol) bSymbol);
+            }
+        }
+
+        Types types = Types.getInstance(new CompilerContext());
+        List<String> relations = new ArrayList<>();
+        int len = bTypeDefinitionSymbols.size();
+        for (int i = 0; i < len; i++) {
+            for (int j = i + 1; j < len; j++) {
+                BTypeDefinitionSymbol td1 = bTypeDefinitionSymbols.get(i);
+                BTypeDefinitionSymbol td2 = bTypeDefinitionSymbols.get(j);
+                BType t1 = td1.type;
+                BType t2 = td2.type;
+                String n1 = td1.name.getValue();
+                String n2 = td2.name.getValue();
+
+                checkSubTypeRelation(t1, t2, n1, n2, types, relations);
+                checkSubTypeRelation(t2, t1, n2, n1, types, relations);
+            }
+        }
+        return relations;
+    }
+
+    private void checkSubTypeRelation(BType sourceType, BType targetType, String sourceTypeName,
+                                      String targetTypeName, Types types, List<String> relations) {
+        if (types.isAssignable(sourceType, targetType)) {
+            relations.add(relation(sourceTypeName, targetTypeName));
+        }
+    }
+
+    private List<String> expectedSubTypeRelations(String filePath) throws IOException {
+        List<String> relations = new ArrayList<>();
+        List<String> lines = Files.readAllLines(resolvePath(filePath));
+        for (String line : lines) {
+            // This check can be changed based on other notations.
+            if (line.startsWith("// ") && line.contains("<:")) {
```

Comment:
Shall we extract these constants to a Constant file?

Reply:
Changed